aten/src/ATen/native/cuda/UnaryFractionKernels.cu

```diff
@@ -29,10 +29,26 @@ void ceil_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+template <typename T>
+__host__ __device__ static inline T frac_wrapper(T a) {
+  if (::isinf(a)) {
```

Comment:
Again, could this use std::modf?

Reply:
Tried to use `std::modf` here for float and double, no significant difference

**Note**: 
* Float16 implementation was unchanged
* `After PR` in the benchmar below is the current state of PR i.e. (a - trunc(a)) approach

```
[------------------------- frac input dtype torch.float16 -------------------------]
                             |  After PR torch.frac CUDA  |  modf PR torch.frac CUDA
1 threads: -------------------------------------------------------------------------
      (128,)                 |             5.2            |             5.3         
      (256, 128)             |             5.1            |             5.3         
      (2, 512, 256)          |             5.1            |             5.2         
      (2, 64, 256, 128)      |            31.6            |            31.6         
      (4, 2, 512, 256, 128)  |           967.9            |           968.0         

Times are in microseconds (us).

[------------------------- frac input dtype torch.float32 -------------------------]
                             |  After PR torch.frac CUDA  |  modf PR torch.frac CUDA
1 threads: -------------------------------------------------------------------------
      (128,)                 |              5.0           |              5.1        
      (256, 128)             |              5.1           |              5.0        
      (2, 512, 256)          |              5.1           |              5.2        
      (2, 64, 256, 128)      |             62.4           |             62.5        
      (4, 2, 512, 256, 128)  |           1943.0           |           1943.0        

Times are in microseconds (us).

[------------------------- frac input dtype torch.float64 -------------------------]
                             |  After PR torch.frac CUDA  |  modf PR torch.frac CUDA
1 threads: -------------------------------------------------------------------------
      (128,)                 |              5.0           |              5.1        
      (256, 128)             |              5.1           |              5.2        
      (2, 512, 256)          |              5.1           |              5.2        
      (2, 64, 256, 128)      |            122.5           |            124.3        
      (4, 2, 512, 256, 128)  |           3874.1           |           3942.7        

Times are in microseconds (us).

```

I don't think it is worth using `std::modf` as we'll have multiple implementation (for half, bfloat using current approach) and `float, double` with std::modf. (Also vectorized version will be based on the current approach)

Let me know how you feel.

Thanks!!